plugins/modules/files/archive.py

```diff
@@ -186,6 +185,34 @@
         HAS_LZMA = False
 
 
+def to_b(s):
+    return to_bytes(s, errors='surrogate_or_strict')
+
+
+def to_n(s):
+    return to_native(s, errors='surrogate_or_strict')
+
+
+def to_na(s):
+    return to_native(s, errors='surrogate_or_strict', encoding='ascii')
+
+
+def expand_paths(paths):
+    expanded_path = []
+    is_globby = []
```

Comment:
Why not make this a boolean with default value `False`, which is set to `True` if a glob is encountered? Would be more efficient.

Reply:
`True`, `True`. Just changed it.